### PR TITLE
make clientcmd use inClusterConfig only if no other configs are available

### LINKED ISSUE #10068: client code, kubecfg file being ignored by kube2sky in favor of service account token
We end up making a non-verified connection to the wrong IP (service IP instead of VM IP).

@lavalamp @roberthbailey @mikedanese

### PR DESCRIPTION
**Pull request by mikedanese**
Closes #10068

### CODE REVIEW

**Comment by thockin** [comment]
Who should review this?

**Comment by mikedanese** [comment]
@jlowdermilk @lavalamp are either of you familiar with this package?

**Comment by mikedanese** [comment]
cc @roberthbailey

**Comment by mikedanese** [comment]
actually hold off on this, I'm going to do a bit more testing.

File: pkg/client/clientcmd/merged_client_builder.go [hunk 1]
```diff
@@ -76,12 +72,23 @@ func (config DeferredLoadingClientConfig) RawConfig() (clientcmdapi.Config, erro
 
 // ClientConfig implements ClientConfig
 func (config DeferredLoadingClientConfig) ClientConfig() (*client.Config, error) {
+
```
Comment (lavalamp) [inline pkg/client/clientcmd/merged_client_builder.go]: nit: extra blank line

**Comment by lavalamp** [comment]
Looks ok-ish to me. @jlowdermilk?

**Comment by mikedanese** [comment]
This seems to be working...
```
I0619 19:52:08.349161       1 kube2sky.go:388] Etcd server found: http://127.0.0.1:4001
I0619 19:52:09.351928       1 kube2sky.go:451] Using https://kubernetes-master for kubernetes master
I0619 19:52:09.352058       1 kube2sky.go:452] Using kubernetes API 
I0619 19:52:09.352597       1 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kube2sky/v0.19.0 (linux/amd64) kubernetes/unknown" -H "Authorization: Bearer Yh1F7Neow3gQVSgXUx1jJyHHjYhBEJK3" https://kubernetes-master/api/v1/endpoints
I0619 19:52:09.352779       1 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kube2sky/v0.19.0 (linux/amd64) kubernetes/unknown" -H "Authorization: Bearer Yh1F7Neow3gQVSgXUx1jJyHHjYhBEJK3" https://kubernetes-master/api/v1/services
I0619 19:52:09.529655       1 debugging.go:120] GET https://kubernetes-master/api/v1/services 200 OK in 176 milliseconds
I0619 19:52:09.529740       1 debugging.go:126] Response Headers:
I0619 19:52:09.529833       1 debugging.go:129]     Content-Type: application/json
I0619 19:52:09.529864       1 debugging.go:129]     Date: Fri, 19 Jun 2015 19:52:09 GMT
I0619 19:52:09.530839       1 request.go:762] Response Body: {"kind":"ServiceList","apiVersion":"v1","metadata":{"selfLink":"/api/v1/services","resourceVersion":"1056"},"items":[{"metadata":{"name":"kube-dns","namespace":"default","selfLink":"/api/v1/namespaces/de
fault/services/kube-dns","uid":"c7948e74-16ba-11e5-bb49-42010af09327","resourceVersion":"139","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"k8s-app":"kube-dns","kubernetes.io/cluster-service":"true","kubernetes.io/name":"KubeDNS"}},"spec":{"ports":[{"name"
:"dns","protocol":"UDP","port":53,"targetPort":53,"nodePort":0},{"name":"dns-tcp","protocol":"TCP","port":53,"targetPort":53,"nodePort":0}],"selector":{"k8s-app":"kube-dns"},"clusterIP":"10.0.0.10","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalanc
er":{}}},{"metadata":{"name":"kubernetes","namespace":"default","selfLink":"/api/v1/namespaces/default/services/kubernetes","uid":"c1c06cd1-16ba-11e5-bb49-42010af09327","resourceVersion":"6","creationTimestamp":"2015-06-19T19:38:27Z","labels":{"component":"apiserv
er","provider":"kubernetes"}},"spec":{"ports":[{"protocol":"TCP","port":443,"targetPort":443,"nodePort":0}],"clusterIP":"10.0.0.1","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}},{"metadata":{"name":"monitoring-grafana","namespace":"defa
ult","selfLink":"/api/v1/namespaces/default/services/monitoring-grafana","uid":"c7598a99-16ba-11e5-bb49-42010af09327","resourceVersion":"126","creationTimestamp":"2015-06-19T19:38:36Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"Grafana"
}},"spec":{"ports":[{"protocol":"TCP","port":80,"targetPort":8080,"nodePort":0}],"selector":{"k8s-app":"influxGrafana"},"clusterIP":"10.0.7.124","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}},{"metadata":{"name":"monitoring-heapster","n
amespace":"default","selfLink":"/api/v1/namespaces/default/services/monitoring-heapster","uid":"c76f8182-16ba-11e5-bb49-42010af09327","resourceVersion":"133","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io
/name":"Heapster"}},"spec":{"ports":[{"protocol":"TCP","port":80,"targetPort":8082,"nodePort":0}],"selector":{"k8s-app":"heapster"},"clusterIP":"10.0.181.156","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}},{"metadata":{"name":"monitorin
g-influxdb","namespace":"default","selfLink":"/api/v1/namespaces/default/services/monitoring-influxdb","uid":"c79ab6c3-16ba-11e5-bb49-42010af09327","resourceVersion":"144","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"kubernetes.io/cluster-service":"true",
"kubernetes.io/name":"InfluxDB"}},"spec":{"ports":[{"name":"http","protocol":"TCP","port":8083,"targetPort":8083,"nodePort":0},{"name":"api","protocol":"TCP","port":8086,"targetPort":8086,"nodePort":0}],"selector":{"k8s-app":"influxGrafana"},"clusterIP":"10.0.249.
87","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}}]}
I0619 19:52:09.532652       1 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kube2sky/v0.19.0 (linux/amd64) kubernetes/unknown" -H "Authorization: Bearer Yh1F7Neow3gQVSgXUx1jJyHHjYhBEJK3" https://kubernetes-master/api/v1/watch/services?resourceVersion=1056
I0619 19:52:09.532850       1 kube2sky.go:245] Setting DNS record: kube-dns.default.cluster.local. -> "{\"host\":\"10.0.0.10\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: kube-dns.default.cluster.local.
I0619 19:52:09.534218       1 debugging.go:120] GET https://kubernetes-master/api/v1/endpoints 200 OK in 181 milliseconds
I0619 19:52:09.534264       1 debugging.go:126] Response Headers:
I0619 19:52:09.534300       1 debugging.go:129]     Content-Type: application/json
I0619 19:52:09.534341       1 debugging.go:129]     Date: Fri, 19 Jun 2015 19:52:09 GMT
I0619 19:52:09.534395       1 request.go:762] Response Body: {"kind":"EndpointsList","apiVersion":"v1","metadata":{"selfLink":"/api/v1/endpoints","resourceVersion":"1056"},"items":[{"metadata":{"name":"kube-dns","namespace":"default","selfLink":"/api/v1/namespaces
/default/endpoints/kube-dns","uid":"c79f1d3e-16ba-11e5-bb49-42010af09327","resourceVersion":"1045","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"k8s-app":"kube-dns","kubernetes.io/cluster-service":"true","kubernetes.io/name":"KubeDNS"}},"subsets":[{"addres
ses":[{"ip":"10.244.1.3","targetRef":{"kind":"Pod","namespace":"default","name":"kube-dns-v3-wif4b","uid":"c759604b-16ba-11e5-bb49-42010af09327","resourceVersion":"1044"}}],"ports":[{"name":"dns-tcp","port":53,"protocol":"TCP"},{"name":"dns","port":53,"protocol":"
UDP"}]}]},{"metadata":{"name":"kubernetes","namespace":"default","selfLink":"/api/v1/namespaces/default/endpoints/kubernetes","uid":"c1c2eb50-16ba-11e5-bb49-42010af09327","resourceVersion":"7","creationTimestamp":"2015-06-19T19:38:27Z"},"subsets":[{"addresses":[{"
ip":"10.240.147.39"}],"ports":[{"port":443,"protocol":"TCP"}]}]},{"metadata":{"name":"monitoring-grafana","namespace":"default","selfLink":"/api/v1/namespaces/default/endpoints/monitoring-grafana","uid":"c7692381-16ba-11e5-bb49-42010af09327","resourceVersion":"833
","creationTimestamp":"2015-06-19T19:38:36Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"Grafana"}},"subsets":[{"addresses":[{"ip":"10.244.2.3","targetRef":{"kind":"Pod","namespace":"default","name":"monitoring-influx-grafana-v1-5zke2","
uid":"c7503617-16ba-11e5-bb49-42010af09327","resourceVersion":"831"}}],"ports":[{"port":8080,"protocol":"TCP"}]}]},{"metadata":{"name":"monitoring-heapster","namespace":"default","selfLink":"/api/v1/namespaces/default/endpoints/monitoring-heapster","uid":"c778d8ad
-16ba-11e5-bb49-42010af09327","resourceVersion":"411","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"Heapster"}},"subsets":[{"addresses":[{"ip":"10.244.0.3","targetRef":{"kind":"Pod","namespace":"d
efault","name":"monitoring-heapster-v3-gunvp","uid":"c73d59c6-16ba-11e5-bb49-42010af09327","resourceVersion":"410"}}],"ports":[{"port":8082,"protocol":"TCP"}]}]},{"metadata":{"name":"monitoring-influxdb","namespace":"default","selfLink":"/api/v1/namespaces/default
/endpoints/monitoring-influxdb","uid":"c7ad298f-16ba-11e5-bb49-42010af09327","resourceVersion":"832","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"InfluxDB"}},"subsets":[{"addresses":[{"ip":"10.24
4.2.3","targetRef":{"kind":"Pod","namespace":"default","name":"monitoring-influx-grafana-v1-5zke2","uid":"c7503617-16ba-11e5-bb49-42010af09327","resourceVersion":"831"}}],"ports":[{"name":"http","port":8083,"protocol":"TCP"},{"name":"api","port":8086,"protocol":"T
CP"}]}]}]}
I0619 19:52:09.535594       1 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kube2sky/v0.19.0 (linux/amd64) kubernetes/unknown" -H "Authorization: Bearer Yh1F7Neow3gQVSgXUx1jJyHHjYhBEJK3" https://kubernetes-master/api/v1/watch/endpoints?resourceVersion=1056
I0619 19:52:09.535850       1 kube2sky.go:193] could not find service for endpoint "kubernetes" in namespace "default"
I0619 19:52:09.535907       1 kube2sky.go:193] could not find service for endpoint "kubernetes" in namespace "default"
I0619 19:52:09.535990       1 kube2sky.go:193] could not find service for endpoint "monitoring-grafana" in namespace "default"
I0619 19:52:09.536258       1 kube2sky.go:193] could not find service for endpoint "monitoring-grafana" in namespace "default"
I0619 19:52:09.536317       1 kube2sky.go:193] could not find service for endpoint "monitoring-heapster" in namespace "default"
I0619 19:52:09.536343       1 kube2sky.go:193] could not find service for endpoint "monitoring-heapster" in namespace "default"
I0619 19:52:09.536402       1 kube2sky.go:193] could not find service for endpoint "monitoring-influxdb" in namespace "default"
I0619 19:52:09.536430       1 kube2sky.go:193] could not find service for endpoint "monitoring-influxdb" in namespace "default"
I0619 19:52:09.536577       1 debugging.go:120] GET https://kubernetes-master/api/v1/watch/services?resourceVersion=1056 200 OK in 3 milliseconds
I0619 19:52:09.536599       1 debugging.go:126] Response Headers:
I0619 19:52:09.536634       1 debugging.go:129]     Date: Fri, 19 Jun 2015 19:52:09 GMT
I0619 19:52:09.536677       1 debugging.go:129]     Content-Type: text/plain; charset=utf-8
I0619 19:52:09.537007       1 debugging.go:120] GET https://kubernetes-master/api/v1/watch/endpoints?resourceVersion=1056 200 OK in 1 milliseconds
I0619 19:52:09.537041       1 debugging.go:126] Response Headers:
I0619 19:52:09.537064       1 debugging.go:129]     Date: Fri, 19 Jun 2015 19:52:09 GMT
I0619 19:52:09.537076       1 debugging.go:129]     Content-Type: text/plain; charset=utf-8
I0619 19:52:09.542130       1 kube2sky.go:245] Setting DNS record: kube-dns.default.svc.cluster.local. -> "{\"host\":\"10.0.0.10\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: 5f106532.kube-dns.default.svc.cluster.local.
I0619 19:52:09.590187       1 kube2sky.go:245] Setting DNS record: kubernetes.default.cluster.local. -> "{\"host\":\"10.0.0.1\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: kubernetes.default.cluster.local.
I0619 19:52:09.606425       1 kube2sky.go:245] Setting DNS record: kubernetes.default.svc.cluster.local. -> "{\"host\":\"10.0.0.1\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: f488b1a2.kubernetes.default.svc.cluster.local.
I0619 19:52:09.622251       1 kube2sky.go:245] Setting DNS record: monitoring-grafana.default.cluster.local. -> "{\"host\":\"10.0.7.124\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: monitoring-grafana.default.cluster.local.
I0619 19:52:09.638581       1 kube2sky.go:245] Setting DNS record: monitoring-grafana.default.svc.cluster.local. -> "{\"host\":\"10.0.7.124\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: 320e8bf.monitoring-grafana.default.svc.cluster.local.
I0619 19:52:09.654259       1 kube2sky.go:245] Setting DNS record: monitoring-heapster.default.cluster.local. -> "{\"host\":\"10.0.181.156\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: monitoring-heapster.default.cluster.local.
I0619 19:52:09.670659       1 kube2sky.go:245] Setting DNS record: monitoring-heapster.default.svc.cluster.local. -> "{\"host\":\"10.0.181.156\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: 6458ad0b.monitoring-heapster.default.svc.cluster.local.
I0619 19:52:09.686308       1 kube2sky.go:245] Setting DNS record: monitoring-influxdb.default.cluster.local. -> "{\"host\":\"10.0.249.87\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: monitoring-influxdb.default.cluster.local.
I0619 19:52:09.702664       1 kube2sky.go:245] Setting DNS record: monitoring-influxdb.default.svc.cluster.local. -> "{\"host\":\"10.0.249.87\",\"priority\":10,\"weight\":10,\"ttl\":30}", with recordKey: 2586a533.monitoring-influxdb.default.svc.cluster.local.
```
and
```
/ # ps axf
PID   USER     COMMAND
    1 root     /kube2sky -domain=cluster.local -kubecfg_file=/etc/dns_token/kub
   10 root     /bin/sh
   17 root     ps axf
/ # cat /etc/dns_token/kubeconfig 
apiVersion: v1
kind: Config
users:
- name: system:dns
  user:
    token: Yh1F7Neow3gQVSgXUx1jJyHHjYhBEJK3
clusters:
- name: local
  cluster:
     server: https://kubernetes-master
     certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURYRENDQWtTZ0F3SUJBZ0lKQU05V1p1VHRULzJkTUEwR0NTcUdTSWIzRFFFQkN3VUFNQ014SVRBZkJnTlYKQkFNVUdERTBOaTR4TkRndU16SXVOamhBTVRRek5EYzBNalV4TmpBZUZ3MHhOVEEyTVRreE9UTTFNVFphRncweQpOVEEyTVRZeE9UTTFNVFphTUNNeElUQWZCZ05WQkFNVUdERTBOaTR4TkRndU16SXVOamhBTVRRek5EYzBNalV4Ck5qQ0NBU0l3RFFZSktvWklodmNOQVFFQkJRQURnZ0VQQURDQ0FRb0NnZ0VCQU14VmNVMXRlTW9mOGR6YUZ5NXQKazlnQWxFb2hEMC90Q0ZjRjY4L0dsRW15SHhDL0dIUjRTY3dmdVZtVmd1Q0RLU1gvaHVRZFV2b1F0cmxxaWpNVgo2VjZPTHdSQWVDWXVvUkowL3pLQjFaeGlyRFh6WFRvM3labnl1YnhoemJtaTRBeHNjZHpEOFlvTENCbVp1Q0RTCjRscnJkb3lQdUxFSXJ5Q2I3VFkyQUJhQWh6c3MyQnhYbjRUQ05XOUJuUWx3VGhXck02ZXl1R0FyYmZoYnBSOVkKVklZWVdBSU16OXRtZjhKWEpxSGtIOERPZmJ3RzMrektialc5bm1kSUNjYWhvZTVxNVcwNlppeVA5ZCtUdTBwWAp3aENEeXZQZnJVRjRtOXFkUzAvaXJ6NUQ0OStqU0VLekRQWllyQmljVVowbEUzTUUwR1F2dWNrRnBTMjU5OFphCnBUTUNBd0VBQWFPQmtqQ0JqekFkQmdOVkhRNEVGZ1FVMVJ2ejhNTjVHbCszWkZveXpIRUZiN3FhYVZZd1V3WUQKVlIwakJFd3dTb0FVMVJ2ejhNTjVHbCszWkZveXpIRUZiN3FhYVZhaEo2UWxNQ014SVRBZkJnTlZCQU1VR0RFMApOaTR4TkRndU16SXVOamhBTVRRek5EYzBNalV4Tm9JSkFNOVdadVR0VC8yZE1Bd0dBMVVkRXdRRk1BTUJBZjh3CkN3WURWUjBQQkFRREFnRUdNQTBHQ1NxR1NJYjNEUUVCQ3dVQUE0SUJBUUJkaUVxbTVlWlhIWFkvWDdhZHh6bFcKbTF6VzFYR1pRN3g2TnRxeTRkelVNeTBtT3FINmw0SkM3dEtHL2FxWEJOWXZjQ0dKUVgyQUl4Z0pnZ3RRQ3pweQpKb3IyYW9KNnU2Z1pzWDVYRjBFMGlwWWJUaG5pZEI4czFuc1hGL3VVY3d2em1hMUtxckJ1a2tKQXRpUU9NTXJiCjR1SkV0L1JCS1RVVll0Z3MwUEpSWmh6enpjVkYwMnUydGVkRFQzRDZ0ZEEzY0hJTldXZjRhbStSeGVzL3hNWkkKeUZkY3JscmU0dklyZ00xd1NNNStQV09VZGw3WGtxWUxGUEI1SklUcFFMcmNkeUxHcG10dXAwUXRUOTVhRy9aRApjMTg3a3RGUGhPQk5jR2ZlQmxqUWRqYUZGL0ZZcC9wcmRpelE0OFA2UUtldHFWYVZvenR4VjBKUGVpQk1hNmRFCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
contexts:
- context:
    cluster: local
    user: system:dns
  name: service-account-context
current-context: service-account-context
```

**Comment by lavalamp** [comment]
Does it break using kubectl in a container, e.g. if you rebuild examples/kubectl-container & then try the example there, does it still work?

**Comment by lavalamp** [comment]
Unfortunately, there's not a good way to check that in an e2e test, because there's no way to get kubectl from head into a container that an e2e test can pull. Yet.

**Comment by mikedanese** [comment]
Here is some logs from running kubectl with this change in that same container:
```
/ # wget http://storage.googleapis.com/public-mikedanese-k8s/kubectl && chmod +x kubectl
Connecting to storage.googleapis.com (173.194.195.128:80)
kubectl              100% |*******************************| 19722k  0:00:00 ETA
/ # ./kubectl --v=10 cluster-info
I0619 20:07:59.349204      39 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kubectl/v0.18.0 (linux/amd64) kubernetes/c93a7d6" -H "Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJkZWZhdWx0Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImRlZmF1bHQtdG9rZW4taHdqZTkiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoiZGVmYXVsdCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMxZmVhYzhhLTE2YmEtMTFlNS1iYjQ5LTQyMDEwYWYwOTMyNyIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpkZWZhdWx0OmRlZmF1bHQifQ.Lj6s1CRk9pee9AsVCstTxvdUTDJzTETywn7nYo0GYtZjvSt-zAgo5JA4VDM3kGz6-v7V3HhtrPKrGQhOKNLagbQTNou31tIDQGD4Q8pM0v0-ZndePzsCfyKk2fizNyW4mxrVs-EpTQdOAwK0HS5fh1I66fXg-T5wjbj5ksn4wmlgECfR1R9ThnOuvlCdAh3GwsqowQNZDvLIYo69MxQw6XE9V-uMVuCnphA-nsAbtqnrox9DVv9zG0wwp_fI_5xbeVEmXlRpuGQBtW6S1E6U6nV_2mzy_uoabfB43RmO-dCOLg91iM7EQ_6EVyqiqkgWPMRK3EHbOYMuR-EvSatIxA" https://10.0.0.1:443/api
I0619 20:07:59.509041      39 debugging.go:120] GET https://10.0.0.1:443/api 200 OK in 159 milliseconds
I0619 20:07:59.509300      39 debugging.go:126] Response Headers:
I0619 20:07:59.509502      39 debugging.go:129]     Content-Type: application/json
I0619 20:07:59.509701      39 debugging.go:129]     Date: Fri, 19 Jun 2015 20:07:59 GMT
I0619 20:07:59.509928      39 debugging.go:129]     Content-Length: 47
I0619 20:07:59.510129      39 request.go:762] Response Body: {
  "versions": [
    "v1beta3",
    "v1"
  ]
}
Kubernetes master is running at https://10.0.0.1:443
I0619 20:07:59.512388      39 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kubectl/v0.18.0 (linux/amd64) kubernetes/c93a7d6" -H "Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJkZWZhdWx0Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImRlZmF1bHQtdG9rZW4taHdqZTkiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoiZGVmYXVsdCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMxZmVhYzhhLTE2YmEtMTFlNS1iYjQ5LTQyMDEwYWYwOTMyNyIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpkZWZhdWx0OmRlZmF1bHQifQ.Lj6s1CRk9pee9AsVCstTxvdUTDJzTETywn7nYo0GYtZjvSt-zAgo5JA4VDM3kGz6-v7V3HhtrPKrGQhOKNLagbQTNou31tIDQGD4Q8pM0v0-ZndePzsCfyKk2fizNyW4mxrVs-EpTQdOAwK0HS5fh1I66fXg-T5wjbj5ksn4wmlgECfR1R9ThnOuvlCdAh3GwsqowQNZDvLIYo69MxQw6XE9V-uMVuCnphA-nsAbtqnrox9DVv9zG0wwp_fI_5xbeVEmXlRpuGQBtW6S1E6U6nV_2mzy_uoabfB43RmO-dCOLg91iM7EQ_6EVyqiqkgWPMRK3EHbOYMuR-EvSatIxA" https://10.0.0.1:443/api
I0619 20:07:59.554301      39 debugging.go:120] GET https://10.0.0.1:443/api 200 OK in 41 milliseconds
I0619 20:07:59.554537      39 debugging.go:126] Response Headers:
I0619 20:07:59.554881      39 debugging.go:129]     Date: Fri, 19 Jun 2015 20:07:59 GMT
I0619 20:07:59.555046      39 debugging.go:129]     Content-Length: 47
I0619 20:07:59.555262      39 debugging.go:129]     Content-Type: application/json
I0619 20:07:59.555355      39 request.go:762] Response Body: {
  "versions": [
    "v1beta3",
    "v1"
  ]
}
I0619 20:07:59.556467      39 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kubectl/v0.18.0 (linux/amd64) kubernetes/c93a7d6" -H "Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJkZWZhdWx0Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImRlZmF1bHQtdG9rZW4taHdqZTkiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoiZGVmYXVsdCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMxZmVhYzhhLTE2YmEtMTFlNS1iYjQ5LTQyMDEwYWYwOTMyNyIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpkZWZhdWx0OmRlZmF1bHQifQ.Lj6s1CRk9pee9AsVCstTxvdUTDJzTETywn7nYo0GYtZjvSt-zAgo5JA4VDM3kGz6-v7V3HhtrPKrGQhOKNLagbQTNou31tIDQGD4Q8pM0v0-ZndePzsCfyKk2fizNyW4mxrVs-EpTQdOAwK0HS5fh1I66fXg-T5wjbj5ksn4wmlgECfR1R9ThnOuvlCdAh3GwsqowQNZDvLIYo69MxQw6XE9V-uMVuCnphA-nsAbtqnrox9DVv9zG0wwp_fI_5xbeVEmXlRpuGQBtW6S1E6U6nV_2mzy_uoabfB43RmO-dCOLg91iM7EQ_6EVyqiqkgWPMRK3EHbOYMuR-EvSatIxA" https://10.0.0.1:443/api/v1beta3/namespaces/default/services?labelSelector=kubernetes.io%2Fcluster-service%3Dtrue
I0619 20:07:59.579797      39 debugging.go:120] GET https://10.0.0.1:443/api/v1beta3/namespaces/default/services?labelSelector=kubernetes.io%2Fcluster-service%3Dtrue 200 OK in 22 milliseconds
I0619 20:07:59.580199      39 debugging.go:126] Response Headers:
I0619 20:07:59.580461      39 debugging.go:129]     Content-Type: application/json
I0619 20:07:59.580785      39 debugging.go:129]     Date: Fri, 19 Jun 2015 20:07:59 GMT
I0619 20:07:59.580950      39 request.go:762] Response Body: {"kind":"ServiceList","apiVersion":"v1beta3","metadata":{"selfLink":"/api/v1beta3/namespaces/default/services","resourceVersion":"1812"},"items":[{"metadata":{"name":"kube-dns","namespace":"default","selfLink":"/api/v1beta3/namespaces/default/services/kube-dns","uid":"c7948e74-16ba-11e5-bb49-42010af09327","resourceVersion":"139","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"k8s-app":"kube-dns","kubernetes.io/cluster-service":"true","kubernetes.io/name":"KubeDNS"}},"spec":{"ports":[{"name":"dns","protocol":"UDP","port":53,"targetPort":53,"nodePort":0},{"name":"dns-tcp","protocol":"TCP","port":53,"targetPort":53,"nodePort":0}],"selector":{"k8s-app":"kube-dns"},"portalIP":"10.0.0.10","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}},{"metadata":{"name":"monitoring-grafana","namespace":"default","selfLink":"/api/v1beta3/namespaces/default/services/monitoring-grafana","uid":"c7598a99-16ba-11e5-bb49-42010af09327","resourceVersion":"126","creationTimestamp":"2015-06-19T19:38:36Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"Grafana"}},"spec":{"ports":[{"protocol":"TCP","port":80,"targetPort":8080,"nodePort":0}],"selector":{"k8s-app":"influxGrafana"},"portalIP":"10.0.7.124","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}},{"metadata":{"name":"monitoring-heapster","namespace":"default","selfLink":"/api/v1beta3/namespaces/default/services/monitoring-heapster","uid":"c76f8182-16ba-11e5-bb49-42010af09327","resourceVersion":"133","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"Heapster"}},"spec":{"ports":[{"protocol":"TCP","port":80,"targetPort":8082,"nodePort":0}],"selector":{"k8s-app":"heapster"},"portalIP":"10.0.181.156","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}},{"metadata":{"name":"monitoring-influxdb","namespace":"default","selfLink":"/api/v1beta3/namespaces/default/services/monitoring-influxdb","uid":"c79ab6c3-16ba-11e5-bb49-42010af09327","resourceVersion":"144","creationTimestamp":"2015-06-19T19:38:37Z","labels":{"kubernetes.io/cluster-service":"true","kubernetes.io/name":"InfluxDB"}},"spec":{"ports":[{"name":"http","protocol":"TCP","port":8083,"targetPort":8083,"nodePort":0},{"name":"api","protocol":"TCP","port":8086,"targetPort":8086,"nodePort":0}],"selector":{"k8s-app":"influxGrafana"},"portalIP":"10.0.249.87","type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}}]}
KubeDNS is running at https://10.0.0.1:443/api/v1beta3/proxy/namespaces/default/services/kube-dns
Grafana is running at https://10.0.0.1:443/api/v1beta3/proxy/namespaces/default/services/monitoring-grafana
Heapster is running at https://10.0.0.1:443/api/v1beta3/proxy/namespaces/default/services/monitoring-heapster
InfluxDB is running at https://10.0.0.1:443/api/v1beta3/proxy/namespaces/default/services/monitoring-influxdb
/ # cat /var/run/secrets/kubernetes.io/serviceaccount/token 
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJkZWZhdWx0Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImRlZmF1bHQtdG9rZW4taHdqZTkiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoiZGVmYXVsdCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMxZmVhYzhhLTE2YmEtMTFlNS1iYjQ5LTQyMDEwYWYwOTMyNyIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpkZWZhdWx0OmRlZmF1bHQifQ.Lj6s1CRk9pee9AsVCstTxvdUTDJzTETywn7nYo0GYtZjvSt-zAgo5JA4VDM3kGz6-v7V3HhtrPKrGQhOKNLagbQTNou31tIDQGD4Q8pM0v0-ZndePzsCfyKk2fizNyW4mxrVs-EpTQdOAwK0HS5fh1I66fXg-T5wjbj5ksn4wmlgECfR1R9ThnOuvlCdAh3GwsqowQNZDvLIYo69MxQw6XE9V-uMVuCnphA-nsAbtqnrox9DVv9zG0wwp_fI_5xbeVEmXlRpuGQBtW6S1E6U6nV_2mzy_uoabfB43RmO-dCOLg91iM7EQ_6EVyqiqkgWPMRK3EHbOYMuR-EvSatIxA 
```

**Comment by mikedanese** [comment]
Tested kubectl-container and everything works as expected when I cherry-pick #10105

**Comment by lavalamp** [comment]
Great! LGTM

**Comment by jlowdermilk** [comment]
LGTM. This looks like the right way to fix defaulting to the less secure kubeconfig.

**Comment by roberthbailey** [comment]
Thanks for the fix @mikedanese!

**Pull request merged**